doc/requirements_and_expectations.md

```diff
@@ -0,0 +1,30 @@
+# Requirements for SQL Query Builder
+
+## 1. Functional Requirements
+
+- Provide an interface for selecting tables from the database.
+- Enable creating and configuring relationships (`JOIN`) between selected tables.
+- Support column selection, adding filters (`WHERE`), setting sorting (`ORDER BY`), and grouping (`GROUP BY`) parameters.
+- Automatically generate SQL queries based on user actions.
```

Comment:
not sure that this is requirement, what kind of actions? if write it like "Generate SQL queries based on user selection" it will be much more clearer.